src/lookup-table/utils/dataTypeUtils.ts

```diff
@@ -0,0 +1,103 @@
+import { mapValues } from 'lodash'
+import { FieldMapType } from '../types'
+import { DateTime } from 'luxon'
+
+export const setDataTypes = (fieldMaps: FieldMapType[], rows: object[]) => {
+  // Determine type
+  const done: string[] = []
+  for (const row of rows) {
+    Object.entries(row).forEach(([key, value], index) => {
+      if (!done.includes(key)) {
+        const type = getType(value)
+        if (type !== null) {
+          fieldMaps[index].dataType = type
+          done.push(key)
+        }
+      }
+    })
+
+    if (done.length === fieldMaps.length) break
+  }
+
+  const fieldMap: { [key: string]: any } = fieldMaps.reduce((acc, curr) => {
+    const name = curr.fieldname
+    const type = curr.dataType
+    return { ...acc, [name]: type }
+  }, {})
```

Comment:
This is just a compact version of the main `fieldMaps` array, with a simple `{ key: type }` object. Makes the subsequent lookup a bit cleaner.